Backend/services/notificationService.js

```diff
@@ -24,14 +24,79 @@ const sendNotification = async (notificationData, io) => {
 
   const isImportant = interactionCount >= threshold;
 
-  const notification = {
-    ...notificationData,
-    isImportant,
-  };
-  const savedNotification = await prisma.notification.create({
-    data: notification,
+  // Check for an existing similar notification
+  const existingNotification = await prisma.notification.findFirst({
+    where: {
+      type: notificationData.type,
+      userId: notificationData.userId,
+      listingId: notificationData.listingId,
+      usernameTarget: notificationData.usernameTarget,
+    },
+    orderBy: { createdAt: "desc" },
   });
 
+  let savedNotification;
+  if (existingNotification) {
+    if (notificationData.type === "LIKE") {
+      // Extract existing users who liked
+      let usersWhoLiked = existingNotification.usernameTarget
+        ? existingNotification.usernameTarget.split(", ")
+        : [];
+
+      // Add the new user who liked
+      if (!usersWhoLiked.includes(notificationData.usernameTarget)) {
```

Comment:
what does usernameTarget represent? if we are finding notifications with the usernameTarget on line 33, in what scenario would usersWholiked not contain usernameTarget?

Reply:
It is to prevent having duplicate usernames in the batched notification
**Scenerio explanation**
First Like:
-User @user1 likes the listing.
-No existing LIKE notification is found.
-A new notification is created with usernameTarget of @user1.
Second Like:
-User @user2 likes the same listing.
-An existing LIKE notification is found with usernameTarget containing @user1.
-The code checks if @user2 is in usersWhoLiked checking if the usernameTarget is in it. Since it's not, @user2 is added to the list.
The notification is updated to include both @user1 and @user2.        It is to prevent allowing duplicate names, because i got some instances where a user liked an item twice, meaning i would get @user1, @user1,@user2 liked the listing